### DISCUSSION TITLE
Is vite setup incompatible with the default Remix App Server?

### DISCUSSION THREAD

**Discussion by justin-hackin** [remix-run/remix#9191] [Q&A] [answered]
The docs suggests avoiding the use of a custom server if possible and I don't need it. However, I can't find an example project using vite that _doesn't_ have a server.js file. Is that required? If so, sounds like something important for the [guide](https://remix.run/docs/en/main/future/vite) on this.

**Comment by justin-hackin** [comment]
Some extra context here: after following the guide, I've had a fair bit of problems running the dev server 

1. The remix devtools crashes `TypeError: Cannot read properties of undefined (reading 'on')
    at configureServer ([project dir]/node_modules/remix-development-tools/dist/index.js:106:20)`
2. When I remove that plugin, I get `TypeError: server.bindCLIShortcuts is not a function
    at dev ([project dir]/node_modules/@remix-run/dev/dist/vite/dev.js:75:10)`

**Comment by justin-hackin** [comment]
It would be nice if the supported stacks would be updated to use vite or at least warn the unsuspecting noob that it's a legacy approach as perhaps this would garner a PR.

**Comment by justin-hackin** [comment]
I figured out that the issues I was encountering were related to and outdated version of vite. 

The boilerplate I derived from had vite installed at v4 (just for vitest presumably). 

Upgrading to vite@^5 fixed my issues and I didn't need a custom express server to run my project. 

Now I'm inspired to squelch the mass of peerDeps mismatches from old packages that work fine so I actually notice these issues from npm 🤦‍♂️

**Reply by kiliman** [reply]
I'm glad you figured it out. I was just going to post that it works fine for me.

Here's a ⚡️ StackBlitz https://stackblitz.com/edit/remix-run-remix-suhjqx

NOTE: https://remix.new will create a default Remix project on StackBlitz using Vite+RAS (Remix App Server)

It uses the default template https://github.com/remix-run/remix/tree/main/templates/remix

**Reply by jordanful** [reply, marked as answer]
Thanks, upgrading Vite to 5x solved this issue for me too.

**Reply by BrianOpteran** [reply]
Also another thanks as upgrading to v5 soved the issue for me as well.
